app/models/filesystem.rb

```diff
@@ -46,7 +46,7 @@ def self.process_sub_xml(xmlNode, path, options = {})
         nh = e.attributes.to_h
 
         nh[:base_name] = nh[:name]
-        nh[:name] = File.join(path, nh[:name])
+        nh[:name] = nh[:fqname] || File.join(path, nh[:name])
```

Comment:
When can `nh[:fqname]` be nil?
Can it be nil when there's a glob in `nh[:name]`?

Reply:
Rich, there's no situation in MD5deep (which sets up the XML) that allows ``nh:[name]`` to be set without also setting ``nh:[fqname]``.  Although I found nothing else in the smartstate gem or in the core repo that creates this XML, this was more of a CYA check.  If you don't think it's necessary I can remove the test.

Reply:
If you expect it to always be set, you should probably use the value unconditionally.
This way, testing can confirm that assumption.